tailscale/data_source_users.go

```diff
@@ -0,0 +1,72 @@
+package tailscale
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	tsclient "github.com/tailscale/tailscale-client-go/v2"
+)
+
+func dataSourceUsers() *schema.Resource {
+	return &schema.Resource{
+		Description: "The users data source describes a list of users in a tailnet",
+		ReadContext: dataSourceUsersRead,
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Optional:    true,
+				Type:        schema.TypeString,
+				Description: "Filters the users list to elements whose type is the provided value.",
```

Comment:
When these are enums, do we typically include the values anywhere in the description? Or do we leave that out to avoid getting out of date?

Reply:
Good catch, I had enums like [this](https://github.com/tailscale/terraform-provider-tailscale/pull/411) on here, but took them out because most of the properties are computed and don't allow validators. However, this is not a computed property.